Problems on PyPy3.11: `TypeError: expected integer, got dict object`

When running the test suite on PyPy3.11 7.3.19:

```pytb
$ pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.11.11[pypy-7.3.19-final], pytest-8.4.1, pluggy-1.6.0
rootdir: /tmp/rich-argparse
configfile: pyproject.toml
testpaths: tests
plugins: cov-6.2.1
collected 287 items                                                                                                                   

tests/test_argparse.py ..................x..................................................................................... [ 36%]
.................................................................................F..............................s......... [ 78%]
tests/test_contrib.py .                                                                      [ 79%]
tests/test_django.py .                                                                       [ 79%]
tests/test_optparse.py ..................x................................sss.....                                              [100%]

============================================================== FAILURES ===============================================================
________________________________________________ test_expand_help_format_specifier[*] _________________________________________________

ct = '*'

    @pytest.mark.parametrize("ct", string.printable)
    def test_expand_help_format_specifier(ct):
        prog = 1 if ct in "cdeEfFgGiouxX*" else "PROG"
        help_formatter = RichHelpFormatter(prog=prog)
        action = Action(["-t"], dest="test", help=f"%(prog){ct}")
        try:
>           expected = help_formatter._expand_help(action)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/test_argparse.py:719: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <rich_argparse._argparse.RichHelpFormatter object at 0x0000558d850ae368>
action = Action(option_strings=['-t'], dest='test', nargs=None, const=None, default=None, type=None, choices=None, required=False, help='%(prog)*', metavar=None)

    def _expand_help(self, action):
        params = dict(vars(action), prog=self._prog)
        for name in list(params):
            if params[name] is SUPPRESS:
                del params[name]
        for name in list(params):
            if hasattr(params[name], '__name__'):
                params[name] = params[name].__name__
        if params.get('choices') is not None:
            choices_str = ', '.join([str(c) for c in params['choices']])
            params['choices'] = choices_str
>       return self._get_help_string(action) % params
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       TypeError: expected integer, got dict object

/usr/lib/pypy3.11/argparse.py:643: TypeError
======================================================= short test summary info =======================================================
FAILED tests/test_argparse.py::test_expand_help_format_specifier[*] - TypeError: expected integer, got dict object
========================================= 1 failed, 280 passed, 4 skipped, 2 xfailed in 3.41s =========================================
```

To reproduce:

```
uv venv -p pypy3.11
uv pip install -r requirements-dev.txt
.venv/bin/pytest
```

Reproduced against main @ 366b0c1091aa2b45b8b0332302c4ba33b4b359fa.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -20,7 +20,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14", "pypy3.11"]
         include:
           - os: windows-latest
             python-version: "3.13"

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,11 @@
 - [GH-155](https://github.com/hamdanal/rich-argparse/issues/155),
   [PR-160](https://github.com/hamdanal/rich-argparse/pull/160)
   Python 3.8 is no longer supported (EOL since 7/10/2024)
+- [GH-164](https://github.com/hamdanal/rich-argparse/issues/164),
+  [PR-165](https://github.com/hamdanal/rich-argparse/pull/165)
+  Add support for PyPy 3.11
+
+## 1.7.1 - 2025-05-25
 
 ### Fixes
 - [PR-162](https://github.com/hamdanal/rich-argparse/pull/162)

--- a/scripts/run-tests
+++ b/scripts/run-tests
@@ -2,6 +2,6 @@
 
 set -euxo pipefail
 
-for python in 3.{9..14}; do
-    uvx --python=${python} --with=. --with-requirements=tests/requirements.txt pytest --cov
+for python in "3.9" "3.10" "3.11" "3.12" "3.13" "3.14" "pypy3.11"; do
+    uvx --isolated --python=${python} --with=. --with-requirements=tests/requirements.txt pytest --cov
 done

--- a/tests/test_argparse.py
+++ b/tests/test_argparse.py
@@ -710,8 +710,8 @@ def test_expand_help_format_specifier(ct):
     action = Action(["-t"], dest="test", help=f"%(prog){ct}")
     try:
         expected = help_formatter._expand_help(action)
-    except ValueError as e:
-        with pytest.raises(ValueError) as exc_info:
+    except (ValueError, TypeError) as e:  # pypy raises TypeError
+        with pytest.raises(type(e)) as exc_info:
             help_formatter._rich_expand_help(action)
         assert exc_info.value.args == e.args
     else:

